src/PopEngine/index.ts

```diff
@@ -163,31 +203,18 @@ export class PopEngine {
         document.body.removeChild(popOver);
       }
       this.addPopToPopStore(groupId, null);
-    }.bind(this), pop.opts.timeToHoverOnPop);
-  }
-
-  public popTopPop(): void {
-    escapeStack.pop();
-  }
 
-  public maybePinOrUnpinPopAndParentPops(target: Element, pin: boolean): void {
-    let groupId = target.getAttribute('popgun-group');
-    let pop = this.getPopFromGroupId(groupId);
-    pop.isPinned = pin;
-    target.setAttribute('pinned-pop', '');
-    let parentPop = this._getParentPop(pop);
-    if (parentPop) {
-      this.maybePinOrUnpinPopAndParentPops(parentPop.targetEl, pin);
-    }
-  }
-
-  public isPopAlreadyOpen(targetElement: Element): boolean {
-    let groupId = targetElement.getAttribute('popgun-group');
-    if (this.getPopFromGroupId(groupId)) {
-      return ((this.getPopFromGroupId(groupId).state === PopStateType.SHOWING) &&
-        (this.getPopFromGroupId(groupId).targetEl === targetElement));
-    }
-    return false;
+      // hide parents
+      if (!!pop.parentPop) {
```

Comment:
```
 +      if (!!pop.parentPop) {
 +        let idx = pop.parentPop.childPops.indexOf(pop);
 +        if (idx !== -1) {
 +          pop.parentPop.childPops.splice(idx, 1);
 +        }
```

As per discussion, does the nested if-statement make sense? If we hit that case, sounds like the chain manager is not updating properly in some scenario.